File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
class Solution {
    boolean solution(String s) {
        boolean answer = true;
        int n = 0;
        for (int i = 0; i < s.length(); i++){
            if (s.charAt(i) == '('){
                n++;
            }else{
                n--;
            }
            if (n < 0){
                answer = false;
                break;
            }
        }
        if (n == 0){
            answer = true;
        }else{
            answer = false;
        }

        return answer;
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    l = []
    
    for i in s:
        if i == "(":
            l.append(i)
        else:
            if len(l) <= 0:
                return False
            l.pop()
    
    if len(l) == 0:
        return True
    else:
        return False